okio/src/main/java/okio/RealBufferedSource.java

```diff
@@ -203,6 +203,10 @@ public RealBufferedSource(Source source) {
     return buffer.readUtf8Line(newline);
   }
 
+  @Override public int readUtf8CodePoint() throws IOException {
+    return buffer.readUtf8CodePoint();
```

Comment:
Done.

Also changed BufferedSourceTest to support a special buffered source that receives data one byte at a time. That found a bug in readFully's exception handling.